### DISCUSSION TITLE
What are some actually decent 'defaults' for configuration values?

### DISCUSSION THREAD

**Discussion by Vektor369** [JarodMica/ai-voice-cloning#140] [Q&A]
I understand that every case is different on every machine. This ain't my first rodeo with training things. But using the settings the program defaults to resulted in a training that was going to take days, and quickly ballooned to 50GB about 10% of the way through before I had to put a stop to it.

So, what would be a decent _**complete**_ set of configuration settings to start with?
For argument's sake, let's just go with the following setup:
English
5 minutes of audio (Yes, yes. I know ideally you want at least 10. Just work with me here.)
Nvidia RTX 3070 with 8GB vram

Thanks in advance.

For reference, here are the settings it defaults to:
Epochs: 500
Learning Rate: 0.00001
Mel LR Ratio: 1
Text LR Ratio: 0.01
Learning Rate Scheme: Multistep
(If Learning Rate Scheme is Cos. Annealing) Learning Rate Restarts: 4
Batch Size: 128
Gradient Accumulation Size: 4
Save Frequency (in epochs): 5
Validation Frequency (in epochs): 5
☐ Half Precision
☒ BitsAndBytes
☐ Validation Enabled
Worker Processes: 2
GPUs: 1
Source Model: ./models/tortoise/autoregressive.pth

**Comment by GregMuck99** [comment]
I feel bad saying this. Your nVidia card will work, but you will be limited by it's onboard memory. I have trained (or attempted training) on a 1080, a 3060, a 4070ti, a 4070ti super and a 4090 (like Jason often uses). I have also observed the ballooning of training time and file size when using the 3060. It's especially frustrating when you have trained overnight and find that the latents calculation step that occurs when you use your model for the first time causes a crash. All that time and effort with no result at all, not even gibberish!

If you are serious and want to avoid frustration, pay ~$800 for a 4070ti super (16GB) or a 4090 (24GB). Gigabyte offers a 4090 for $1500. As for the settings, I have gotten good results with 200 epochs, 64 batch size and 8 gradient accumulation. All my other settings are the same as yours. _Assuming_ you are using a good dataset (see my earlier response to you), it's easy to overtrain a model thinking that more epochs must make a better model, right? Jason made a video showing that overtraining can produce noisy, screechie  output. You can select an earlier model in the settings tab which was created during the training process (the number of them determined on your Save Frequency). If you have overtrained, an earlier model (smaller pth file number in settings) will produce better audio than a later one. Weird huh?
